src/pages/studyView/StudyViewPageStore.ts

```diff
@@ -1035,6 +1176,11 @@ export class StudyViewPageStore {
                 this.studyIds = studyIds;
             }
         }
+        if (query.sharedGroups) {
+            this.sharedGroupSet = stringListToSet(
```

Comment:
can you add a comment explaining that this can be done in constructor because it comes from the URL